lib/compiler_rt/memcpy.zig

```diff
@@ -8,21 +9,283 @@ comptime {
     }
 }
 
-pub fn memcpy(noalias dest: ?[*]u8, noalias src: ?[*]const u8, len: usize) callconv(.C) ?[*]u8 {
-    @setRuntimeSafety(false);
+const CopyType = if (std.simd.suggestVectorLength(u8)) |vec_size|
+    @Type(.{ .vector = .{
+        .child = u8,
+        .len = vec_size,
+    } })
+else
+    usize;
+
+const copy_alignment = @alignOf(CopyType);
+const copy_size = @sizeOf(CopyType);
+
+comptime {
+    assert(copy_size >= copy_alignment);
+    assert(std.math.isPowerOfTwo(copy_size));
+}
+
+pub const memcpy = if (builtin.mode == .ReleaseSmall)
+    memcpy_small
+else
+    memcpy_fast;
```

Comment:
this logic belongs in the comptime block where it decides which function to export